content/blog/tidymodels-2021-q2/index.Rmd

```diff
@@ -0,0 +1,232 @@
+  labs(x = NULL)
+```
+
+For some kinds of models, we would want to create dummy or indicator variables from nominal predictors and preprocess features to be on the same scale. We can use recipes for this task:
+
+```{r}
+churn_rec <- 
+  recipe(churn ~ voice_mail_plan + total_intl_minutes + 
+           total_day_minutes + total_eve_minutes + state, 
+         data = churn_train) %>%
+  step_dummy(all_nominal_predictors()) %>%
+  step_normalize(all_predictors())
+```
+
+You can `prep(churn_rec)` to estimate the quantities needed to create categorical features and to scale all the predictors:
+
+```{r}
+churn_prep <- prep(churn_rec)
+churn_prep
+```
+
+To remove everything from this prepped recipe not needed for applying to new data (e.g. [bake()](https://recipes.tidymodels.org/reference/bake.html) it), we can call `butcher(churn_prep)`. In some applications, modeling practitioners need to make custom functions with a feature engineering recipe. Sometimes those functions have... a lot of extra STUFF in them, stuff that is not needed for prediction.
+
+```{r}
+butchered_plus <- function() {
+  some_junk_in_the_environment <- runif(1e6)
+  
+  churn_prep <- 
+    recipe(churn ~ voice_mail_plan + total_intl_minutes + 
+             total_day_minutes + total_eve_minutes + state, 
+           data = churn_train) %>%
+    step_dummy(all_nominal_predictors()) %>%
+    step_normalize(all_predictors()) %>%
+    prep()
+  
+  butcher(churn_prep)
+}
+```
+
+In the old version of butcher, we did not successfully remove all that extra stuff, and recipes were bigger than they needed to be:
+
+```{r, eval=FALSE}
+lobstr::obj_size(butcher(churn_prep))
+#> 1,835,512 B
+lobstr::obj_size(butchered_plus())
+#> 9,836,480 B
+```
+
+In the new version of butcher, we now successfully remove unneeded components from the recipe, so it is smaller:
+
+```{r, eval=FALSE}
+lobstr::obj_size(butcher(churn_prep))
+#> 1,695,352 B
+lobstr::obj_size(butchered_plus())
+#> 1,695,352 B
+```
+
+There are also `butcher()` methods for `workflow()` objects, so when you `butcher()` a modeling workflow, you remove everything not needed for prediction from both its estimated recipe *and* its trained model, making it as lightweight as possible for deployment.
+
+## SVMs and fast logistic regression with LiblineaR
+
+Unfortunately, the `"liquidSVM"` engine for support vector machine models that parsnip supported was deprecated in the latest release, because that package was removed from CRAN. We added a new engine in the same release that allows users to fit linear SVMs with the [parsnip model `svm_linear()`](https://parsnip.tidymodels.org/reference/svm_linear.html), as well as having another option for logistic regression. This new `"LiblineaR"` engine is based on the same C++ library that is shipped with scikit-learn. We'd like to thank the [maintainers of the LiblineaR R package](https://www.dnalytics.com/software/liblinear/) for all their help as we set up this integration.
```

Comment:
Add link to scikit-learn or the relevant C++ lib? (Totally your call—I think it's fine either way).